### PR TITLE
Improving parseUrl() to handle IPv6 and other special URLs

### PR DESCRIPTION
**Pull request by jancurn**
Actually, looking at it, it might be best to just get rid of `parseUrl` completely and use `new URL()`, it only adds `scheme` to the parsed object which can be easily replaced by `protocol` around the code base

### CODE REVIEW

**Comment by valekjo** [comment]
Sadly the `new URL` does more and there are some edge cases where it behaves slightly differently. The fix passes tests, parseUrl and redactUrl tests were updated.

File: src/tools.js [hunk 1]
```diff
@@ -64,31 +64,34 @@ export const isInvalidHeader = (name, value) => {
...
+    };
 
-    if (parsed.auth) {
-        const matches = /^([^:]+)(:?)(.*)$/.exec(parsed.auth);
-        if (matches && matches.length === 4) {
-            parsed.username = matches[1];
-            if (matches[2] === ':') parsed.password = matches[3];
-        }
-    }
+    console.dir({ parsed });
```
Comment (jancurn) [inline src/tools.js]: pls can we get rid of this?

**Review by jancurn — COMMENTED** [review]
Looks good, thanks @valekjo
